scipy/linalg/decomp_svd.py

```diff
@@ -264,35 +264,76 @@ def diagsvd(s, M, N):
 
 # Orthonormal decomposition
 
-def orth(A):
+def orth(A, rcond=None):
     """
     Construct an orthonormal basis for the range of A using SVD
 
     Parameters
     ----------
     A : (M, N) array_like
         Input array
+    rcond : float, optional
+        Relative condition number. Singular values ``s`` smaller than
+        ``rcond * max(s)`` are considered zero.
+        Default: floating point eps * max(M,N).
 
     Returns
     -------
     Q : (M, K) ndarray
         Orthonormal basis for the range of A.
-        K = effective rank of A, as determined by automatic cutoff
+        K = effective rank of A, as determined by rcond
 
     See also
     --------
     svd : Singular value decomposition of a matrix
+    null_space : Matrix null space
 
     """
     u, s, vh = svd(A, full_matrices=False)
     M, N = A.shape
-    eps = numpy.finfo(float).eps
-    tol = max(M, N) * numpy.amax(s) * eps
+    if rcond is None:
+        rcond = numpy.finfo(float).eps * max(M, N)
+    tol = numpy.amax(s) * rcond
     num = numpy.sum(s > tol, dtype=int)
     Q = u[:, :num]
     return Q
 
 
+def null_space(A, rcond=None):
+    """
+    Construct an orthonormal basis for the null space of A using SVD
+
+    Parameters
+    ----------
+    A : (M, N) array_like
+        Input array
+    rcond : float, optional
+        Relative condition number. Singular values ``s`` smaller than
+        ``rcond * max(s)`` are considered zero.
+        Default: floating point eps * max(M,N).
+
+    Returns
+    -------
+    Q : (M, K) ndarray
+        Orthonormal basis for the null space of A.
+        K = effective rank of A, as determined by rcond
+
+    See also
+    --------
+    svd : Singular value decomposition of a matrix
+    orth : Matrix range
+
+    """
+    u, s, vh = svd(A, full_matrices=True)
+    M, N = A.shape
```

Comment:
the input is array_like but it gets different errors for `np.empty((0,0))`, `np.array([])`, `[1]` in `svd` . Also it will err out with list object not having a shape

Reply:
Fixed (input error handling is done by svd).